### ISSUE TITLE
Option to disable the display of Arch news

### ISSUE THREAD

**Issue by leyyce** [Antiz96/arch-update#281]
### Description

Add a config option to disable the display of Arch news

### Motivation and context

`paru` has an built-in option to display new Arch news on it's own. As I sometimes use paru itself without relying on `arch-update` I'd like to enable that option within `paru`. To make sure that the news aren't displayed two times when using `arch-update`, I'd like to have the option to disable it.

**Comment by leyyce** [comment]
Gone ahead and tried to add the necessary changes myself.  If you think this works, I would appreciate if you could merge the pull request 😄

**Comment by Antiz96** [comment]
Hi,

Thanks a lot for the suggestion and the related pull request!

I had mixed feelings about allowing users to disable the news prompt at first, as it's an important step of upgrading the system. But I reckon that people might use different methods than the one built-in in `arch-update` to read Arch news already, resulting in eventual undesired duplicates. So it feels like a fair change after all.

I'll take a look at your pull request soon :wink: 
Thanks again, I appreciate the effort :smile:

**Comment by leyyce** [comment]
Alright thanks 😄 Also please excuse the 4 single commits for each file but I did it in the GitHub only editor really quick 😆

**Comment by Antiz96** [comment]
No problem, don't worry. I'm squashing commits when merging anyway ;)

**Comment by Antiz96** [comment]
I'll give your PR a review tomorrow :)

**Comment by leyyce** [comment]
Hi, thanks for the feedback 😄

To get this out of the way first, there isn't really any specific reason that I sometimes run paru directly instead of just using arch-update. It just happens sometimes, maybe it's because I'm already in the terminal, maybe it's habit, I really don't know, all I know is that it just happens without me thinking about it sometimes ^^

I somehow missed that paru was only used if AUR updates are available. In that case it indeed would be undesirable to deactivate the news for arch-update. On the other hand, like you said, you really don't know what other options some users may be already using outside of arch-update or not (there for example is [informant](https://aur.archlinux.org/packages/informant) which adds a hook to pacman to also check for updates and works for both pacman and paru.

I'd even go as far as to say that for some users and use cases, where the user is not always relying on arch-update, it makes it more convenient and more unlikely that the user misses an important update if he can deactivate the Arch news in arch-update, because otherwise he might decide to just not bother with activating the functionality in paru or pacman because it's inconvenient.

In the end, I think the awesome thing about Linux is that the user has all the choices, and he can use that freedom however he likes no matter if it's a good idea or not. Just like with the Arch news display. There definitely are valid reasons where a user would want to disable them in arch-news, just like there are cases where it doesn't make sense. Anyway, I'd much prefer to leave the ultimate choice of doing so or not to the user.

Of course the default should always be that it's activated and there should be a disclaimer that reading the Arch news is an important step of the update process, and that you should only disable it if you know what you're doing. 

If you wanted to, I could add the necessary disclaimer to the readme (maybe it also would be a good idea to add comments to all the config options about what they're doing into the example config file while already at it?) and maybe even add an output like

> [arch-update] Warning: Arch news checking is disabled.

to the full-upgrade script. But of course ultimately the choice is up to you.

On the other hand, is there a specific reason why, if detected, you're not just using paru regardless of AUR updates being available? I'd have to further look into this, but the needed code paths checking for the presence of paru and probably even using it to see if there are updates available already have to be there anyway right? So it probably should be easy to make it always use paru if present I guess, for example by using some variable like `pacman-frontend` instead of just hard coding it in.

**Comment by Antiz96** [comment]
> Hi, thanks for the feedback 😄
> 

You're welcome :slightly_smiling_face: 

> To get this out of the way first, there isn't really any specific reason that I sometimes run paru directly instead of just using arch-update. It just happens sometimes, maybe it's because I'm already in the terminal, maybe it's habit, I really don't know, all I know is that it just happens without me thinking about it sometimes ^^
> 

Habits are hard to change for sure :stuck_out_tongue_closed_eyes:
Just in case, `arch-update` can also be ran from the terminal though :)

> I somehow missed that paru was only used if AUR updates are available. In that case it indeed would be undesirable to deactivate the news for arch-update. On the other hand, like you said, you really don't know what other options some users may be already using outside of arch-update or not (there for example is [informant](https://aur.archlinux.org/packages/informant) which adds a hook to pacman to also check for updates and works for both pacman and paru.
> 
> I'd even go as far as to say that for some users and use cases, where the user is not always relying on arch-update, it makes it more convenient and more unlikely that the user misses an important update if he can deactivate the Arch news in arch-update, because otherwise he might decide to just not bother with activating the functionality in paru or pacman because it's inconvenient.
> 
> In the end, I think the awesome thing about Linux is that the user has all the choices, and he can use that freedom however he likes no matter if it's a good idea or not. Just like with the Arch news display. There definitely are valid reasons where a user would want to disable them in arch-update, just like there are cases where it doesn't make sense. Anyway, I'd much prefer to leave the ultimate choice of doing so or not to the user.
> 

While I get the point, I don't fully agree with the reasoning here... Arch-Update has been made as an "all in one" tool so that you don't **need to** use anything else (e.g. informant), either by wrapping said tools or providing its own implementation. Now, sure, Linux is all about choices and stuff, so people are obviously free to use whatever they want aside of Arch-Update if they feel like it. 
However, I think it's also fair to say that if someone's using external tools that duplicate with some built-in Arch-Update's core features, then it isn't really an Arch-Update matter in my opinion? I think "I'm already using X to do Y, so I'd like Y to be de-activatable from Arch-Update" is not really a valid reasoning after all... While I can't prevent people to use whatever they like (nor do I want to), Arch-Update has been made to be a "all in one" & "self-sufficient" tool. Arch-Update is designed to work with itself, if people are only using it partially (as in, switching solutions from time to time to cover the same need), I don't think this is an Arch-Update problem.

> Of course the default should always be that it's activated and there should be a disclaimer that reading the Arch news is an important step of the update process, and that you should only disable it if you know what you're doing.
> 
> If you wanted to, I could add the necessary disclaimer to the readme (maybe it also would be a good idea to add comments to all the config options about what they're doing into the example config file while already at it?) and maybe even add an output like
> 
> > [arch-update] Warning: Arch news checking is disabled. It's highly recommended to enable it if you're not already relying on another way to check them.
> 
> to the full-upgrade script. That would still clutter the output a bit, but at least doesn't need interaction from the user anymore. But of course ultimately the choice is up to you.
> 

That would be a potential solution, but I'm unsure how I feel about merging an option that I don't recommend from the start...

> On the other hand, is there a specific reason why, if detected, you're not just using paru regardless of AUR updates being available? I'd have to further look into this, but the needed code paths checking for the presence of paru (or other AUR helpers) and probably even using it to see if there are updates available already have to be there anyway right? So it probably should be easy to make it always use paru if present I guess, for example by using some variable like `pacman-frontend` instead of just hard coding it in or by removing the "AUR update available" check from the AUR update script and adding a "AUR helper present" check to the "normal" update script. I'd have to look at what AUR helpers you're supporting again, but all of them should be compatible with the pacman flags I think.

Yes there's a reason why.
AUR helpers / pacman wrappers are not officially supported and are subject to "breakages" when e.g. a new pacman / libalpm version is out (as in, they should be made compatible with the said new version upstream wise and should be *manually* rebuilt by users since such AUR helpers won't be able to upgrade themselves after a pacman / libalpm major update). For instance, it took several days for `paru` to release a version compatible with pacman 7, resulting in it being unusable as-is for some time. Note that I'm not accusing anyone of anything, I'm just exposing why AUR helpers are unsupported by Arch and shouldn't be used when there's no need to in my opinion.
This is why Arch-Update will only use AUR helpers for AUR packages and keep using plain official pacman for the rest. No need to introduce an extra layer that could be a source of potential issue if the said extra layer isn't required.

To sum up, I think I got carried away by the fact that you took the time to make an implementation effort (which I sincerely appreciate :heart:), but the truth is I'm not on par with allowing some steps to be disabled completely (whether it's for news reading or not).
Arch-Update has been designed to follow **every** recommended system maintenance steps, as described in the [Arch Wiki](https://wiki.archlinux.org/title/System_maintenance), to provide a complete upgrade path. Allowing to disabled some of those steps altogether kinda defeats the purpose (specifically if it is to do the same thing with another tool :sweat_smile:) and could be a source of potential issue.
When it comes to news reading specifically, *only* 8 Arch news have been published in 2024. I think it's a reasonable amount of eventual duplicate news printing for people relying on additional external method for news tracking. Regardless, I'd better be on the safe side by making sure people are aware of the news, rather than leaving the door opened to people missing it (by allowing them to disable it entirely).

**Comment by Antiz96** [comment]
Once again, I sincerely appreciate your effort! But, after a second thought, I am not on par with this change...  

As said above, Arch-Update has been made to cover **every** steps needed for a proper and complete upgrade. I don't feel comfortable allowing to disable some of them completely as it kinda defeats the purpose. When it comes to news reading specifically, I don't think the actual benefit that disabling it would bring for such specific cases is worth the eventual issues it could imply (namely missing news), especially knowing the low number of news posted each year.

I do understand that people might eventually use other tools that could duplicate with some Arch-Update's features, but I think it's fair to say that this is not an Arch-Update problem. 
Arch-Update has been designed so you don't need to use such other tools. If you still want to use them in parallel anyway, then that's your decision.

As such, I'll put this issue as `wontfix`, sorry about that...
But, again, thanks anyway for the suggestion and your overall interest toward this project! :heart:

**Comment by leyyce** [comment]
Alright, I totally understand if you don't want to add this as it goes against the philosophy of your software. And don't worry, only took me a few minutes anyway so it's not much work done that's being tossed away 😄

Like you said, it's just a minor inconvenience anyway, but I thought it wouldn't be hard to add that option, so I at least tried. 

Moving forward I now have a few options that still kind of seem acceptable and solve the problem:
- just accept the duplicate news display like you said as, it doesn't happen that often anyway
- Give arch-update a nice short alias and try to add running that instead of paru into my muscle memory
- Just keep the changes I made in my fork and install that instead
- Try what happens if I set the amount of news to display to zero 😱 (and hope that if it works you won't patch it out 😆)

> In case you're still interested in contributing, there's another issue currently opened that should be fairly easy to tackle though :angel: 🤗

I'll definitely take a look at it, probably tomorrow. Would love to help to add something to an awesome tool like arch-update, even if it's just something small like that, thanks for maintaining it, it's much appreciated ❤️

**Comment by Antiz96** [comment]
> Alright, I totally understand if you don't want to add this as it goes against the philosophy of your software. And don't worry, only took me a few minutes anyway so it's not much work done that's being tossed away 😄
> 
> Like you said, it's just a minor inconvenience anyway, but I thought it wouldn't be hard to add that option, so I at least tried.
> 

Thanks for being comprehensive :)

> Moving forward I now have a few options that still kind of seem acceptable and solve the problem:
> 
>     * just accept the duplicate news display like you said as, it doesn't happen that often anyway
> 
>     * Give arch-update a nice short alias and try to add running that instead of paru into my muscle memory
> 
>     * Just keep the changes I made in my fork and install that instead
> 
>     * Try what happens if I set the amount of news to display to zero 😱 (and hope that if it works you won't patch it out 😆)
> 

Solution 1 and / or 2 feels the most straightforward :P But I hope you'll be able to find the option that suits you the most!
Just so you know, setting the amount of news to 0 won't work, the code expects a positive number.

> 
> > In case you're still interested in contributing, there's another issue currently opened that should be fairly easy to tackle though 👼 🤗
> 
> I'll definitely take a look at it, probably tomorrow. Would love to help to ~add~ remove something ~to~ from an awesome tool like arch-update 😆, 

Sure, feel free to give it a go if you want to, that's appreciated :)

> even if it's just something small like that, thanks for maintaining it, it's much appreciated ❤️

And that means the world, thanks a lot for the kind words and the support <3
